src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/FileSystem.Windows.vb

```diff
@@ -0,0 +1,204 @@
+﻿Imports System
+Imports System.ComponentModel
+Imports System.Diagnostics
+Imports Microsoft.VisualBasic.CompilerServices
+Namespace Microsoft.VisualBasic.FileIO
+    Partial Public Class FileSystem
+        ''' <summary>
+        ''' Sets relevant flags on the SHFILEOPSTRUCT and calls SHFileOperation to copy move file / directory.
+        ''' </summary>
+        ''' <param name="Operation">Copy or move.</param>
+        ''' <param name="TargetType">The target is a file or directory?</param>
+        ''' <param name="FullSourcePath">Full path to source directory / file.</param>
+        ''' <param name="FullTargetPath">Full path to target directory / file.</param>
+        ''' <param name="ShowUI">Show all dialogs or just the error dialogs.</param>
+        ''' <param name="OnUserCancel">Throw exception or ignore if user cancels the operation.</param>
+        ''' <remarks>
+        ''' Copy/MoveFile will call this directly. Copy/MoveDirectory will call ShellCopyOrMoveDirectory first
+        ''' to change the path if needed.
+        ''' </remarks>
+        Private Shared Sub ShellCopyOrMove(ByVal Operation As CopyOrMove, ByVal TargetType As FileOrDirectory,
+            ByVal FullSourcePath As String, ByVal FullTargetPath As String, ByVal ShowUI As UIOptionInternal, ByVal OnUserCancel As UICancelOption)
+            Debug.Assert(System.Enum.IsDefined(GetType(CopyOrMove), Operation))
+            Debug.Assert(System.Enum.IsDefined(GetType(FileOrDirectory), TargetType))
+            Debug.Assert(FullSourcePath.Length <> 0 AndAlso IO.Path.IsPathRooted(FullSourcePath), "Invalid FullSourcePath!!!")
+            Debug.Assert(FullTargetPath.Length <> 0 AndAlso IO.Path.IsPathRooted(FullTargetPath), "Invalid FullTargetPath!!!")
+            Debug.Assert(ShowUI <> UIOptionInternal.NoUI, "Why call ShellDelete if ShowUI is NoUI???")
```

Comment:
Doesn't ShellDelete allow the deleted file to be put into the recycle bin where a direct delete operation won't?

Reply:
@Wraith2 The code in this file and all the src files is original, I am just writing tests and making minimal changes to be consistent with CoreFx and to split for OS specific issues. The Assert is in the original code.

Reply:
@Wraith2 actually some of the Asserts are wrong in the original source but I guess were never tested. TargetPath is tested for Length<>0 but it is valid to be Nothing in the case of a Delete operation.